server/src/endpoints/qs/ws/dispatch.rs

```diff
@@ -2,44 +2,93 @@
 //
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-use super::{
-    InternalQsWsMessage,
-    messages::{Connect, Disconnect, NotifyMessage, NotifyMessageError},
-};
+use std::collections::HashMap;
+
 use actix::{
     ResponseFuture,
     prelude::{Actor, Context, Handler, Recipient},
 };
-use phnxtypes::identifiers::QsClientId;
+use phnxprotos::{
+    convert::RefInto,
+    queue_service::v1::{ListenResponse, QueueEvent, QueueUpdate, listen_response},
+};
+use phnxtypes::{
+    identifiers::QsClientId,
+    messages::client_ds::{DsEventMessage, QsWsMessage},
+};
+use tokio::sync::mpsc;
+use tracing::info;
 
-use std::collections::HashMap;
+use super::{
+    InternalQsWsMessage,
+    messages::{Connect, Disconnect, GrpcConnect, NotifyMessage, NotifyMessageError},
+};
 
 enum NotifyClientError {
     ClientNotFound,
 }
 
+enum DispatchDestination {
+    Actor(Recipient<InternalQsWsMessage>),
```

Comment:
This is to preserve the original Websocket functionality, correct?

Reply:
Yes, I did not remove it: neither in the backend nor on the client.